src/main/scala/ch/openolitor/stammdaten/StammdatenInsertService.scala

```diff
@@ -473,6 +476,74 @@ class StammdatenInsertService(override val sysConfig: SystemConfig) extends Even
     }
   }
 
+  def createZusatzAbo(meta: EventMetadata, newId: AboId, create: ZusatzAboCreate)(implicit personId: PersonId = meta.originator) = {
+    DB localTxPostPublish { implicit session => implicit publisher =>
+      val hauptAbo = aboById(create.hauptAboId)
+      val zusatzAbotyp = zusatzAboTypById(create.abotypId)
+      hauptAbo match {
+        case Some(h) => {
+          zusatzAbotyp match {
+            case Some(z) => {
+              val startDate = defaultDateStart(h.start, z.aktivVon)
+              val endDate = defaultDateEnd(h.ende, z.aktivBis)
+              val zusatzAbo = copyTo[ZusatzAboCreate, ZusatzAbo](
+                create,
+                "id" -> newId,
+                "hauptAbotypId" -> h.abotypId,
+                "kunde" -> h.kunde,
+                "vertriebsartId" -> h.vertriebsartId,
+                "vertriebId" -> h.vertriebId,
+                "vertriebBeschrieb" -> h.vertriebBeschrieb,
+                "abotypName" -> z.name,
+                "start" -> startDate,
+                "ende" -> endDate,
+                "guthabenVertraglich" -> h.guthabenVertraglich,
+                "guthaben" -> h.guthaben,
+                "guthabenInRechnung" -> h.guthabenInRechnung,
+                "letzteLieferung" -> h.letzteLieferung,
+                "anzahlAbwesenheiten" -> h.anzahlAbwesenheiten,
+                "anzahlLieferungen" -> h.anzahlLieferungen,
+                "aktiv" -> h.aktiv,
+                "erstelldat" -> meta.timestamp,
+                "ersteller" -> meta.originator,
+                "modifidat" -> meta.timestamp,
+                "modifikator" -> meta.originator
+              )
+              DB autoCommitSinglePublish { implicit session => implicit publisher =>
+                stammdatenWriteRepository.insertEntity[ZusatzAbo, AboId](zusatzAbo)
+              }
+            }
+            case None => throw new RuntimeException("The id provided does not corresponde to any zusatzabotyp");
+          }
+        }
+        case None => throw new RuntimeException("The id provided does not corresponde to any zusatzabo");
+      }
+    }
+  }
+
+  def defaultDateStart(date1: LocalDate, date2: Option[LocalDate]): LocalDate = {
+    date2 match {
+      case Some(d2) => {
+        if ((date1 compareTo d2) > 0) date1 else d2
+      }
+      case None => date1
+    }
+  }
+
```

Comment:
Maybe something like that is easier to comprehend?

def defaultDateStart2(date1: LocalDate, date2: Option[LocalDate]): LocalDate = {
  date2.map { d2 =>
    if ((date1 compareTo d2) > 0) date1 else d2
  }.getOrElse(date1)
}

Reply:
Or even:
```
def defaultDateStart2(date1: LocalDate, date2: Option[LocalDate]): LocalDate = {
 (date1, date2) match {
      case (d1, None) => d1
      case (d1, Some(d2)) => d2 if (d1 compareTo d2) > 0 )> d1
      case (_, d2) => d2
    }    
}
```

Then you see all the decisions in one level.

Reply:
I like yours Mike, I'll change it !